compiler/aarch64/codegen/OMRTreeEvaluator.cpp

```diff
@@ -6146,6 +6146,74 @@ TR::Register *commonLoadEvaluator(TR::Node *node, TR::InstOpCode::Mnemonic op, i
    return commonLoadEvaluator(node, op, size, tempReg, cg);
    }
 
+static TR::Register *
+loadAddress(TR::Node *node, TR::MemoryReference *mr, TR::CodeGenerator *cg)
+   {
+   TR::Register *targetReg = cg->allocateRegister();
+
+   // TODO: this probably exists somewhere else
+   TR::Register *baseReg = mr->getBaseRegister();
+   TR::Register *indexReg = mr->getIndexRegister();
+   intptr_t offset = mr->getOffset();
+
+   if (offset == 0)
+      {
+      if (baseReg != NULL && indexReg == NULL)
+         {
+         if (baseReg != targetReg)
+            generateMovInstruction(cg, node, targetReg, baseReg);
+         }
+      else if (indexReg != NULL && baseReg == NULL)
+         {
+         if (baseReg != targetReg)
+            generateMovInstruction(cg, node, targetReg, indexReg);
+         }
+      else
+         {
+         generateTrg1Src2Instruction(cg, TR::InstOpCode::addx, node, targetReg, baseReg, indexReg);
+         }
```

Comment:
You need to consider the cases where the `indexReg` is scaled or extended.
`TR::MemoryReference::createWithIndexReg()` has `scale` and `extendCode` in the parameter list.
https://github.com/eclipse-omr/omr/blob/e711cd83b2bfd56cb81da10413b0035c34e5c553/compiler/aarch64/codegen/MemoryReference.hpp#L107